### DISCUSSION TITLE
[과목 1] 제 1장 데이터 모델링의 이해 연습문제

### DISCUSSION THREAD

**Discussion by white-gyu** [flataex/sqlp-study#1] [연습문제] [answered]
### 연습 문제

**문제 1. 다음 중 정보시스템을 모델링할 때 세 가지 관점에 해당하지 않는 것은?**
① 업무가 어떤 데이터와 관련이 있는지 분석
② 업무가 실제하는 일은 무엇인지 또는 무엇을 해야 하는지 분석
③ 업무가 처리하는 일의 방법에 따라 데이터가 어떻게 영향을 받는지 분석
④ 업무를 처리할 수 있는 프로그램 구성을 어떻게 해야 하는지 분석

**문제 2. 데이터 모델링의 세 가지 중요개념에 속하지 않는 것은?**
① 업무가 관여하는 어떤 것(Things)
② 업무가 관여하는 어떤 것의 행위(Events)
③ 업무가 관여하는 어떤 것의 성격(Attributes)
④ 업무가 관여하는 어떤 것의 관계(Relationships)

**문제 3. 발생시점에 따라 구분할 수 있는 엔터티의 유형이 아닌 것은?**
① 행위엔터티(Active Entity)
② 중심엔터티(Main Entity)
③ 개념엔터티(Conceptual Entity)
④ 기본엔터티(Basic Entity)

**문제 4. 이 속성이 없어도 다른 속성을 이용하여 결과를 도출할 수 있는 특징을 가진 속성의 이름은?** 
① 설계속성(Designed Attribute)
② 파생속성(Derived Attribute)
③ 기본속성(Basic Attribute)
④ 관계속성(Associative Attribute)

**문제 5. 다음 중 엔터티의 특징에 포함되지 않는 것은?**
① 반드시 해당 업무에서 필요하고 관리하고자 하는 정보여야 한다.
② 유일한 식별자로 식별할 수 있어야 한다.
③ 엔터티는 업무 프로세스에 의해 이용되어야 한다.
④ 엔터티는 반드시 속성이 없어도 된다.

**문제 6. 다음 설명이 나타내는 데이터 모델의 개념은 무엇인가?**
> 학생이라는 엔터티가 있을 때 학점이라는 속성 값의 범위는 0.0에서 4.0 사이의 실수 값이며, 주소라는
속성은 길이가 20자리 이내의 문자열로 정의할 수 있다.

① 도메인(Domain)
② 용어사전(Word Dictionary)
③ 속성사전(Attribute Dictionary)
④ 시스템카탈로그(System Catalog)

**문제 7. 엔터티간 1:1, 1:M과 같이 관계의 기수성을 나타내는 것을 무엇이라 하는가?** 
① 관계명(Relationship Membership)
② 관계차수(Relationship Degree/Cardinality)
③ 관계선택성(Relationship Optionality)
④ 관계정의(Relationship Definition)

**문제 8. 관계를 정의할 때 주요하게 체크해야 하는 사항과 거리가 먼 것은?**
① 두 개의 엔터티 사이에 관심 있는 연관규칙이 존재하는가?
② 두 개의 엔터티 사이에 정보의 조합이 발생되는가?
③ 업무기술서, 장표에 관계연결에 대한 규칙이 서술되어 있는가?
④ 업무기술서, 장표에 관계연결을 가능하게 하는 명사(Noun)가 있는가?

**문제 9. 식별자의 대체 여부에 따라 분류하는 방식은?** 
① 주식별자 - 보조식별자
② 내부식별자- 외부식별자 
③ 본질식별자- 인조식별자
④ 단일식별자 - 복합식별자

**문제 10. 다음 개념에 해당하는 관계는 어떤 관계를 설명한 것인가?**
> 부모엔터티로부터 속성을 받았지만 자식엔터티의 주식별자로 사용하지 않고 일반적인 속성으로만 사용
한다.

① 식별자관계(Identifying Relationship)
② 일반속성관계(Attribute Relationship)
③ 비식별자관계(Non-Identifying Relationship)
④ 외부식별관계(Foreign Key Relationship)

**Comment by white-gyu** [comment, marked as answer]
### 연습 문제

**문제 1. 다음 중 정보시스템을 모델링할 때 세 가지 관점에 해당하지 않는 것은?**
① 업무가 어떤 데이터와 관련이 있는지 분석
② 업무가 실제하는 일은 무엇인지 또는 무엇을 해야 하는지 분석
③ 업무가 처리하는 일의 방법에 따라 데이터가 어떻게 영향을 받는지 분석
④ 업무를 처리할 수 있는 프로그램 구성을 어떻게 해야 하는지 분석

> ④

**문제 2. 데이터 모델링의 세 가지 중요개념에 속하지 않는 것은?**
① 업무가 관여하는 어떤 것(Things)
② 업무가 관여하는 어떤 것의 행위(Events)
③ 업무가 관여하는 어떤 것의 성격(Attributes)
④ 업무가 관여하는 어떤 것의 관계(Relationships)

> ②

**문제 3. 발생시점에 따라 구분할 수 있는 엔터티의 유형이 아닌 것은?**
① 행위엔터티(Active Entity)
② 중심엔터티(Main Entity)
③ 개념엔터티(Conceptual Entity)
④ 기본엔터티(Basic Entity)

> ③, 개념 엔티티는 [데이터 모델링의 3단계 진행](https://github.com/flataex/sqlp-study/tree/bghan/%EC%A0%9C01%EC%9E%A5%20%EB%8D%B0%EC%9D%B4%ED%84%B0%20%EB%AA%A8%EB%8D%B8%EB%A7%81%EC%9D%98%20%EC%9D%B4%ED%95%B4/bghan#4-%EB%8D%B0%EC%9D%B4%ED%84%B0-%EB%AA%A8%EB%8D%B8%EB%A7%81%EC%9D%98-3%EB%8B%A8%EA%B3%84-%EC%A7%84%ED%96%89) 에 해당

**문제 4. 이 속성이 없어도 다른 속성을 이용하여 결과를 도출할 수 있는 특징을 가진 속성의 이름은?** 
① 설계속성(Designed Attribute)
② 파생속성(Derived Attribute)
③ 기본속성(Basic Attribute)
④ 관계속성(Associative Attribute)

> ②

**문제 5. 다음 중 엔터티의 특징에 포함되지 않는 것은?**
① 반드시 해당 업무에서 필요하고 관리하고자 하는 정보여야 한다.
② 유일한 식별자로 식별할 수 있어야 한다.
③ 엔터티는 업무 프로세스에 의해 이용되어야 한다.
④ 엔터티는 반드시 속성이 없어도 된다.

> ④, 테이블 내 필드는 있어야 함

**문제 6. 다음 설명이 나타내는 데이터 모델의 개념은 무엇인가?**
> 학생이라는 엔터티가 있을 때 학점이라는 속성 값의 범위는 0.0에서 4.0 사이의 실수 값이며, 주소라는
속성은 길이가 20자리 이내의 문자열로 정의할 수 있다.

① 도메인(Domain)
② 용어사전(Word Dictionary)
③ 속성사전(Attribute Dictionary)
④ 시스템카탈로그(System Catalog)

> ①, 도메인 제약 조건 설명

**문제 7. 엔터티간 1:1, 1:M과 같이 관계의 기수성을 나타내는 것을 무엇이라 하는가?** 
① 관계명(Relationship Membership)
② 관계차수(Relationship Degree/Cardinality)
③ 관계선택성(Relationship Optionality)
④ 관계정의(Relationship Definition)

> ②

**문제 8. 관계를 정의할 때 주요하게 체크해야 하는 사항과 거리가 먼 것은?**
① 두 개의 엔터티 사이에 관심 있는 연관규칙이 존재하는가?
② 두 개의 엔터티 사이에 정보의 조합이 발생되는가?
③ 업무기술서, 장표에 관계연결에 대한 규칙이 서술되어 있는가?
④ 업무기술서, 장표에 관계연결을 가능하게 하는 명사(Noun)가 있는가?

> ④, has/contains 같은 동사

**문제 9. 식별자의 대체 여부에 따라 분류하는 방식은?** 
① 주식별자 - 보조식별자
② 내부식별자- 외부식별자 
③ 본질식별자- 인조식별자
④ 단일식별자 - 복합식별자

> ③

**문제 10. 다음 개념에 해당하는 관계는 어떤 관계를 설명한 것인가?**
> 부모엔터티로부터 속성을 받았지만 자식엔터티의 주식별자로 사용하지 않고 일반적인 속성으로만 사용
한다.

① 식별자관계(Identifying Relationship)
② 일반속성관계(Attribute Relationship)
③ 비식별자관계(Non-Identifying Relationship)
④ 외부식별관계(Foreign Key Relationship)

> ③

**Comment by beolggoolosori** [comment]
문제 1: ④ 업무를 처리할 수 있는 프로그램 구성을 어떻게 해야 하는지 분석
문제 2: ② 업무가 관여하는 어떤 것의 행위(Events)
문제 3: ③ 개념 엔터티
문제 4: ② 파생속성(Derived Attribute)
문제 5: ④ 엔터티는 반드시 속성이 없어도 된다.
문제 6: ① 도메인(Domain)
문제 7: ② 관계차수(Relationship Degree/Cardinality)
문제 8: ④ 업무기술서, 장표에 관계연결을 가능하게 하는 명사(Noun)가 있는가?
문제 9: ③ 본질식별자- 인조식별자
문제 10: ③ 비식별자관계(Non-Identifying Relationship)
